### PR TITLE
Double loop Notebooks compatible w/ idaes 2.0

### PR DESCRIPTION
**Pull request by dguittet**
## Addresses issue:
Part 2 of https://github.com/gmlc-dispatches/dispatches/issues/154

## Summary/Motivation:
Update a notebook for idaes 2.0

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md and COPYRIGHT.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
Summarizing a few points from the dev call:

- [ ] Remove explicit version checks (in both the notebook(s) and the Python modules)
- [ ] Modify `setup.py` so that the `idaes-pse` requirement points to the soon-to-be-merged IDAES/idaes-pse#986
- [ ] Modify `.github/workflows/notebook-checks.yml` so that `DoubleLoopOptimization.ipynb` is not excluded from the CI job that executes that notebooks

**Comment by dguittet** [comment]
thanks @lbianchi-lbl this branch is now passing with the tests using https://github.com/IDAES/idaes-pse/pull/986

that PR is approved and ready to merge @ksbeattie and after it is merged, I will switch this PR to use the main idaes branch not the PR branch

**Comment by lbianchi-lbl** [comment]
@dguittet sounds good. I've just merged IDAES/idaes-pse#986, and I'll shortly be creating a tag on the `gmlc-dispatches/idaes-pse` fork. Once that's available, I'll let you know so that you can modify `setup.py` to point to that for the `idaes-pse` prerelease requirement.

File: setup.py
Comment (lbianchi-lbl) [inline setup.py:71]:
```suggestion
        "idaes-pse @ https://github.com/gmlc-dispatches/idaes-pse/archive/2.0.0.dev4.dispatches.22.10.26.zip"
```

@dguittet I've just created the `2.0.0.dev4.dispatches.22.10.26` tag corresponding to `IDAES/idaes-pse:main` up to and including (the merge commit for) IDAES/idaes-pse#986.

Comment (dguittet) [reply setup.py]: @lbianchi-lbl there is yet another fix we needed in IDAES https://github.com/IDAES/idaes-pse/pull/996 to get all the double loop notebooks to work. could you please merge that one and make another tag? This PR is now being re-tested with the idaes branch above, but I will switch to the dispatches/idaes tag

Comment (lbianchi-lbl) [reply setup.py]: @dguittet I've just created `2.0.0.dev4.dispatches.22.10.26.1` that includes IDAES/idaes-pse#996.

Comment (dguittet) [reply setup.py]: Thanks! updated

**Comment by dguittet** [comment]
All the double loop notebooks are running with the latest IDAES updates. The RE and NE ones solve to an optimal bidding solution. 

The FE one doesn't, so in this PR, I reduced the size of the problem by reducing the bidding horizon to 4 hours. IPOPT reaches its max 200 limit iterations in 15 minutes. Since the notebook was already failing, I don't think this new development is a problem, and actually is an improvement by reducing the time to failure. 

@radhakrishnatg @nareshsusarla could you please review the changes here to the double loop notebooks?

**Comment by lbianchi-lbl** [comment]
@dguittet Everything looks good for my side. I've just noticed that now you managed to fix all the discrepancies, we could try to see if the RE `DoubleLoopOptimization.ipynb` notebook works without skipping cells if we add the `DISPATCHES_TESTING_MODE` env var to the CI environment. We could also do this in another PR (and arguably it would be good to get this merged in sooner rather than later), so I leave the decision to you.

**Review by radhakrishnatg — APPROVED** [review]
Thank you @dguittet  for making changes to the nuclear case study notebook. Everything looks good. Just want to make sure that the only change to the code is renaming `GeneratorModelData` to `ThermalGeneratorModelData`. Correct? Are there any other changes to the code?

**Comment by dguittet** [comment]
@radhakrishnatg Yes, pretty much. Thanks for the review.

@lbianchi-lbl Good point. I'd prefer another PR, just so this one can get finished.

File: dispatches/case_studies/renewables_case/wind_battery_double_loop.py
Comment (dguittet) [reply dispatches/case_studies/renewables_case/wind_battery_double_loop.py:163]: Ah ok thanks for the info

**Review by lbianchi-lbl — APPROVED** [review]

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl Good point. I'd prefer another PR, just so this one can get finished.

Sure, I agree. Should I go ahead and merge it, or you prefer to get an all-clear on the NE DL notebook first?

**Comment by dguittet** [comment]
Merging would be great, thanks

**Pull request merged**
